lib/rules/no-unstable-nested-components.js

```diff
@@ -24,11 +23,11 @@ const HOOK_REGEXP = /^use[A-Z0-9].*$/;
 
 /**
  * Generate error message with given parent component name
- * @param {String} parentName Name of the parent component
+ * @param {String} parentName Name of the parent component, if known
  * @returns {String} Error message with parent component name
  */
 function generateErrorMessageWithParentName(parentName) {
-  return `Declare this component outside parent component "${parentName}" or memoize it.`;
+  return `Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component${parentName ? ` “${parentName}” ` : ' '}and pass data as props.`;
```

Comment:
can this be passed in with `data` instead of generating a different template literal every time?

Reply:
on further thought, it could, but it'd make the conditional space trickier, so we won't